xs/closure.c

```diff
@@ -266,6 +266,10 @@ ffi_pl_closure_call(ffi_cif *ffi_cif, void *result, void **arguments, void *user
       case FFI_PL_TYPE_OPAQUE:
         *((void**)result) = SvOK(sv) ? INT2PTR(void*, SvIV(sv)) : NULL;
         break;
+      case FFI_PL_TYPE_RECORD_VALUE:
+        /* TODO: type checking! */
```

Comment:
What to do if Perl returns us something other than a record value?

 - Warn unconditionally.  We can unfortunately not really die, but this is almost certainly a bug in your code.
 - Fill in the return struct with all zeros.